packages/atomic/src/components/result-template-components/result-template-decorators.tsx

```diff
@@ -81,11 +83,11 @@ const resultContextEventName = 'atomic/resolveResult';
  * @param element The element that the event is dispatched to, which must be the child of a rendered "atomic-result".
  * @returns A promise that resolves on initialization of the parent "atomic-result" element, or rejects when there is no parent "atomic-result" element.
  */
-export const resultContext = (element: Element) =>
-  new Promise<FoldedResult | Result>((resolve, reject) => {
-    const event = buildCustomEvent<ResultContextEventHandler>(
+export function resultContext<T = Result>(element: Element) {
```

Comment:
We should constrain it
```suggestion
export function resultContext<T extends Result | FoldedResult = Result>(element: Element) {
```

Reply:
woaaah nice